### PR TITLE
Do_weld proc

### PR DESCRIPTION
**Pull request by MadmanMartian**
Adds the do_weld proc, bringing the action of welding onto the welder itself.

This means less checks thrown around in the code of differring quality, and now the alternate welding sound is being used.

Also adds a weld_speed variable to welders. Higher weld_speed factor, quicker welding.

Also changes (hopefully all of) the remaining istype(var, /obj/item/weapon/weldingtool)) to iswelder(var)

### CODE REVIEW

**Comment by ShiftyRail** [comment]
>Do_welp

This proc should be called everytime you look at the codebase

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
you've been visited by the MADMAN MARTIAN of SHITCODE DESTRUCTION

**Comment by Kurfursten** [comment]
I never thought someone would have the will to do this

**Comment by spoonylove** [comment]
So gatling welders will be fast now?

**Comment by MadmanMartian** [comment]
They can be, but they aren't right now. PR is too unatomic to have a PoC in it

**Comment by Exxion** [comment]
Technically there's already a system for specifying tool speeds but it sucks and should be replaced with something more like this

File: code/ATMOSPHERICS/components/unary/vent_pump.dm [hunk 1]
```diff
@@ -302,29 +302,22 @@
...
-					welded = 1
-					update_icon()
-				else
-					user.visible_message("[user] unwelds the vent.", "You unweld the vent.", "You hear welding.")
-					investigation_log(I_ATMOS, "has been unwelded by [user.real_name] ([formatPlayerPanel(user, user.ckey)]) at [formatJumpTo(get_turf(src))]")
-					welded = 0
-					update_icon()
+		to_chat(user, "<span class='notice'>Now welding the vent.</span>")
+		if (WT.do_weld(user, src, 20, 1))
+			if(gcDestroyed)
```
Comment (unid15) [inline code/ATMOSPHERICS/components/unary/vent_pump.dm:309]: Is this not checked on the last tick of do_weld?

**Pull request merged**